thalerj/openbmctool.py

```diff
@@ -0,0 +1,1490 @@
+        print("Failed to create zip file with collected information")
+    return "data collection complete"
+
+"""
+     handles various bmc level commands, currently bmc rebooting
+       
+     @param host: string, the hostname or IP address of the bmc
+     @param args: contains additional arguments used by the bmc sub command
+     @param session: the active session to use
+     @param args.json: boolean, if this flag is set to true, the output will be provided in json format for programmatic consumption 
+"""         
+def bmc(host, args, session):
+    if(args.info):
+        return "To be completed"
+    if(args.type is not None):
+        return bmcReset(host, args, session)
+    
+"""
+     controls resetting the bmc. warm reset reboots the bmc, cold reset removes the configuration and reboots. 
+       
+     @param host: string, the hostname or IP address of the bmc
+     @param args: contains additional arguments used by the bmcReset sub command
+     @param session: the active session to use
+     @param args.json: boolean, if this flag is set to true, the output will be provided in json format for programmatic consumption 
+""" 
+def bmcReset(host, args, session):
+    if(args.type == "warm"):
+        print("\nAttempting to reboot the BMC...:")
+        url="https://"+host+"/xyz/openbmc_project/state/bmc0/attr/RequestedBMCTransition"
+        httpHeader = {'Content-Type':'application/json'}
+        data = '{"data":"xyz.openbmc_project.State.BMC.Transition.Reboot"'
+        res = session.post(url, headers=httpHeader, data=data, verify=False, timeout=20)
+        return res.text
+    elif(args.type =="cold"):
+        return "cold reset not available at this time."
+    else:
+        return "invalid command"
+"""
+     creates the parser for the command line along with help for each command and subcommand
+       
+     @return: returns the parser for the command line
+""" 
+def createCommandParser():
+    parser = argparse.ArgumentParser(description='Process arguments')
+    parser.add_argument("-H", "--host", required=True, help='A hostname or IP for the BMC')
+    parser.add_argument("-U", "--user", required=True, help='The username to login with')
+    #parser.add_argument("-v", "--verbose", help='provides more detail')
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-A", "--askpw", action='store_true', help='prompt for password')
+    group.add_argument("-P", "--PW", help='Provide the password in-line')
+    parser.add_argument('-j', '--json', action='store_true', help='output json data only')
+    parser.add_argument('-t', '--policyTableLoc', help='The location of the policy table to parse alerts')
+    parser.add_argument('-c', '--CerFormat', action='store_true', help=argparse.SUPPRESS)
+    parser.add_argument('-T', '--procTime', action='store_true', help= argparse.SUPPRESS)
+    subparsers = parser.add_subparsers(title='subcommands', description='valid subcommands',help="sub-command help")
```

Comment:
Lots of inconsistency with long option capitalisation above. I think it would be better to avoid capitalisation altogether and instead separate words with a `-`.